op-node/rollup/derive/engine_queue.go

```diff
@@ -91,6 +120,8 @@ type EngineQueue struct {
 	l1Fetcher L1Fetcher
 }
 
+var _ EngineControl = (*EngineQueue)(nil)
```

Comment:
Maybe something to document about the engine queue is that it should not be used concurrently, had the thought after reading `resetBuildingState`

Reply:
That applies to the whole `derive` package. Not sure what the right place for that is. Can add a warning if it helps.